File: 백준/Bronze/1157. 단어 공부/단어 공부.rs
use std::collections::HashMap;
use std::io::stdin;

fn main() {
    let word = input_line();
    let word = word
        .chars()
        .map(|c| c.to_ascii_uppercase())
        .collect::<String>();

    let mut count_map: HashMap<char, usize> = HashMap::new();
    let mut max_count = usize::MIN;

    for c in word.chars() {
        let count = count_map.entry(c).or_insert(0);
        *count += 1;
        max_count = max_count.max(*count);
    }

    let vec = count_map
        .iter()
        .filter(|(_, count)| **count == max_count)
        .map(|(ch, _)| *ch)
        .collect::<Vec<_>>();

    let ans = if vec.len() == 1 { vec[0] } else { '?' };

    println!("{ans}");
}

fn input_line() -> String {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();
    buf.trim().to_owned()
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.cc
#include <iostream>

using namespace std;

int main() {

  string str;
  cin >> str;

  int count[26] = {
      0,
  };

  for (int i = 0; i < str.length(); ++i) {
    if (str[i] >= 'a') {
      str[i] -= 'a' - 'A';
    }
    count[str[i] - 'A']++;
  }

  int max_count = 0;
  char answer;

  for (int i = 0; i < 26; ++i) {
    if (count[i] == max_count) {
      answer = '?';
    }
    if (count[i] > max_count) {
      max_count = count[i];
      answer = i + 'A';
    }
  }

  cout << answer;

  return 0;
}